explainaboard/processors/kg_link_tail_prediction.py

```diff
@@ -177,6 +181,34 @@ def _statistics_func(self, samples: Iterator[dict[str, str]]):
             "tail_fre": dict_tail,
         }
 
+    def _gen_metric_stats(
+        self, sys_info: SysOutputInfo, sys_output: list[dict]
+    ) -> list[MetricStats]:
+        """Generate sufficient statistics for scoring different metrics.
+
+        :param sys_info: Information about the system outputs
+        :param sys_output: The system output itself
+        :return: Statistics sufficient for scoring
+        """
+
+        metrics = unwrap(self._get_metrics(sys_info))
+        true_data = [self._get_true_label(x) for x in sys_output]
+        pred_data = [self._get_predicted_label(x) for x in sys_output]
+        rank_data = [
+            self._get_rank_data(x) for x in sys_output
+        ]  # rank of true entity in predictions
+        assert all(
+            item is not None for item in rank_data
+        ), 'Some data points do not have rank information; check your system outptus.'
+
+        metric_stats = []
+        for metric in metrics:
+            if isinstance(metric, MeanReciprocalRank) or isinstance(metric, MeanRank):
+                metric_stats.append(metric.calc_stats_from_rank(rank_data))
+            else:
+                metric_stats.append(metric.calc_stats_from_data(true_data, pred_data))
```

Comment:
Hits can also be calculated from rank right? Probably all of them should be calculated from the rank to be consistent and simplify the code?